src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -279,5 +307,61 @@ public boolean delete() {
         // TODO: Add journey state deletion.
         return Persistence.monitoredTrips.removeById(this.id);
     }
+
+    public Map<String, String> parseQueryParams() throws URISyntaxException {
+        return URLEncodedUtils.parse(
+            new URI(String.format("http://example.com/plan?%s", queryParams)),
+            UTF_8
+        ).stream().collect(Collectors.toMap(NameValuePair::getName, NameValuePair::getValue));
+    }
+
+    public boolean isArriveBy() throws URISyntaxException {
+        // if arriveBy is not included in query params, OTP will default to false, so initialize to false
+        return parseQueryParams().getOrDefault("arriveBy", "false").equals("true");
+    }
```

Comment:
I find the code in parseQueryParams and isArriveBy to be a little hard to read and overly clever. I would prefer the use of one or two variables to improve legibility. Also, these could use some basic javadoc.

Reply:
I also agree that the parseQueryParams is clever, but don't think it's too clever. I think breaking it up wouldn't really add much value. I went ahead and added JavaDoc though in 6fd00a976c063c5b46e7254a995fbbb628b90445